gui/src-tauri/src/main.rs

```diff
@@ -286,3 +300,15 @@ fn valid_java_paths() -> Vec<(OsString, i32)> {
 		})
 		.collect()
 }
+
+fn executable<W: AsRef<OsStr> + ?Sized>(bin: &W) -> OsString {
```

Comment:
We don't need to be so generic here. Just accept an `&OsStr`. These bounds over complicate things.